wsmaster/che-core-api-factory-github/src/main/java/org/eclipse/che/api/factory/server/github/GithubAuthorizingFileContentProvider.java

```diff
@@ -51,4 +69,113 @@ protected boolean isPublicRepository(GithubUrl remoteFactoryUrl) {
       return false;
     }
   }
+
+  @Override
+  protected String fetchContent(String fileURL, boolean skipAuthentication)
+      throws IOException, DevfileException {
+
+    final String rawFileURL = formatUrl(fileURL);
+
+    try {
+      if (skipAuthentication) {
+        return urlFetcher.fetch(branchToSHA(rawFileURL, null));
```

Comment:
Why not to extend the `rawFileLocation()` instead of overriding the parent method?

Reply:
I mean here: https://github.com/eclipse-che/che-server/blob/40598a0e8bb532ded283cf4c2471a93d8398b116/wsmaster/che-core-api-factory-github/src/main/java/org/eclipse/che/api/factory/server/github/GithubUrl.java#L182-L196

Reply:
I was trying to do it the first, but the logic of `GithubUrl` became overloaded. I think it's more a question of taste.

There are additional error handlers were added to `GithubAuthorizingFileContentProvider::fetchContent` as well.

Reply:
> There are additional error handlers were added to GithubAuthorizingFileContentProvider::fetchContent as well.

You can define a new `fetchContent()` function with needed parameters
>  I think it's more a question of taste

I don't think so, there is a duplicated piece of code.

Reply:
I would not keep this logic with relacing branch on SHA in [GithubUrl.java](https://github.com/eclipse-che/che-server/blob/40598a0e8bb532ded283cf4c2471a93d8398b116/wsmaster/che-core-api-factory-github/src/main/java/org/eclipse/che/api/factory/server/github/GithubUrl.java#L182-L196).

It's better then to keep it in [GithubURLParser.java](https://github.com/eclipse-che/che-server/blob/40598a0e8bb532ded283cf4c2471a93d8398b116/wsmaster/che-core-api-factory-github/src/main/java/org/eclipse/che/api/factory/server/github/GithubURLParser.java#L136-L175) like we do with pull requests.

Reply:
Reworked